File: Application/src/main/java/com/shareyourproxy/api/rx/event/SyncAllContactsSuccessEvent.java
package com.shareyourproxy.api.rx.event;

import android.os.Parcel;

import com.shareyourproxy.api.rx.command.eventcallback.EventCallback;

/**
 * Sync all users command has successfully completed.
 */
public class SyncAllContactsSuccessEvent extends EventCallback {
    public static final Creator<SyncAllContactsSuccessEvent> CREATOR =
        new Creator<SyncAllContactsSuccessEvent>() {
            @Override
            public SyncAllContactsSuccessEvent createFromParcel(Parcel in) {
                return new SyncAllContactsSuccessEvent();
            }

            @Override
            public SyncAllContactsSuccessEvent[] newArray(int size) {
                return new SyncAllContactsSuccessEvent[size];
            }
        };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/event/SyncAllContactsSuccessEvent.kt
package com.shareyourproxy.api.rx.event

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.rx.command.eventcallback.EventCallback

/**
 * Sync all users command has successfully completed.
 */
class SyncAllContactsSuccessEvent : EventCallback() {

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
    }

    companion object {
        val CREATOR: Parcelable.Creator<SyncAllContactsSuccessEvent> = object : Parcelable.Creator<SyncAllContactsSuccessEvent> {
            override fun createFromParcel(parcel: Parcel): SyncAllContactsSuccessEvent {
                return SyncAllContactsSuccessEvent()
            }

            override fun newArray(size: Int): Array<SyncAllContactsSuccessEvent?> {
                return arrayOfNulls(size)
            }
        }
    }
}
